### PR TITLE
CircleCI: Switch to new master build pipeline

### PR DESCRIPTION
**Pull request by aknuds1**
**What this PR does / why we need it**:
Switch to new [Grafana Build Pipeline](https://github.com/grafana/build-pipeline) based pipeline for building the master branch (and publishing corresponding packages and Docker images).

**Special notes for your reviewer**:
This PR is intended to elicit discussion around the new pipeline, and hopefully uncover any remaining bugs/omissions. You can see the status of the testing of the different packages and Docker images built with the new pipeline in the [test plan](https://docs.google.com/document/d/1BRR2OU3DWCesCRebd-1PJQCTn0nth5lh4RsLiNK2V3Y/edit#).

I know the new CircleCI configuration is imperfect and duplication could be reduced f.ex., but my thinking is to concentrate on making it work first and after the switch we can make more PRs to simplify.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -1896,108 +1728,6 @@ jobs:
...
-            - grafana-docker-master
-          filters: *filter-only-master
-      - release-next-packages:
-          requires:
-            - build-all
-            - test-frontend
-          filters: *filter-only-master
-
-  # Pipeline for building master branch, based on Grafana Build Pipeline tool
   build-pipeline:
```
Comment (marefr) [inline .circleci/config.yml]: Nit, seems like this is only used for master so maybe change the name to denote master?

**Review by marefr — COMMENTED** [review]
Changes looks solid to me. Great work! Added a minor nit/comment. 

Would be nice if we or someone else in the community could help out testing the ones you haven't tested @aknuds1 like:
grafana-6.7.0-5ccebe13pre.armhfp.rpm
grafana-6.7.0-5ccebe13pre.linux-armv6.tar.gz
grafana-enterprise-6.7.0-5ccebe13pre.armhfp.rpm
grafana-enterprise-6.7.0-5ccebe13pre.linux-armv6.tar.gz

Not sure what kind of device you need to test these, is it old Raspberry PI?

File: .circleci/config.yml (hunk 1 above)
Comment (aknuds1) [reply .circleci/config.yml]: @marefr Good observation, but my next step is to try to apply the same workflow for PR and release pipelines also (through branching logic) :)

**Comment by aknuds1** [comment]
> Changes looks solid to me. Great work! Added a minor nit/comment.
> 
> Would be nice if we or someone else in the community could help out testing the ones you haven't tested @aknuds1 like:
> grafana-6.7.0-5ccebe13pre.armhfp.rpm
> grafana-6.7.0-5ccebe13pre.linux-armv6.tar.gz
> grafana-enterprise-6.7.0-5ccebe13pre.armhfp.rpm
> grafana-enterprise-6.7.0-5ccebe13pre.linux-armv6.tar.gz
> 
> Not sure what kind of device you need to test these, is it old Raspberry PI?

Thanks @marefr! I failed to test armhfp RPMs since yum segfaulted when trying to install them, FYI. About ARM v6 I don't remember what stopped me, I probably couldn't find Docker images for that arch.

**Review by xlson — APPROVED** [review]
LGTM!

Great work @aknuds1 :)

**Comment by aknuds1** [comment]
> LGTM!
> 
> Great work @aknuds1 :)

Thanks @xlson!

File: .circleci/config.yml (hunk 1 above)
Comment (marefr) [reply .circleci/config.yml]: Ohh nice makes perfect sense. Please disregard this then for now.

Comment (aknuds1) [reply .circleci/config.yml]: OK :) I don't know for sure yet if I can use a single workflow as I hope, but will try.

**Review by ryantxu — APPROVED** [review]
🎉 exciting to see this... next we see what it breaks!

File: .circleci/config.yml [hunk 2]
```diff
@@ -558,21 +557,21 @@ jobs:
...
           name: Publish Docker manifest
           command: |
             export DOCKER_CLI_EXPERIMENTAL=enabled
-            docker manifest create "grafana/grafana:master-new-pipeline" \
-              "grafana/grafana:master-new-pipeline" \
-              "grafana/grafana-arm32v7-linux:master-new-pipeline" \
-              "grafana/grafana-arm64v8-linux:master-new-pipeline"
-            docker manifest push "grafana/grafana:master-new-pipeline"
+            docker manifest create "grafana/grafana:master" \
+	      "grafana/grafana:master" \
```
Comment (papagian) [inline .circleci/config.yml]: The workflow is failing because of indentation of this line.

Comment (aknuds1) [reply .circleci/config.yml]: @papagian Ah oops, thanks.

**Review by dprokop — APPROVED** [review]
Exciting! 🍾

**Pull request merged**